packages/adblocker-electron-preload/src/index.ts

```diff
@@ -7,24 +7,28 @@
  */
 
 import { ipcRenderer } from 'electron';
+import { DOMMonitor, IBackgroundCallback } from '@cliqz/adblocker-content';
 
-import {
-  DOMMonitor,
-  IBackgroundCallback,
-  IMessageFromBackground,
-  injectScript,
-} from '@cliqz/adblocker-content';
+type CosmeticFiltersResponse = {
+  active: boolean;
+  error?: string;
+};
 
-function getCosmeticsFiltersFirst(): string[] | null {
-  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
-}
-function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
+function injectCosmeticFilters(
+  data?: Omit<IBackgroundCallback, 'lifecycle'>,
+): Promise<CosmeticFiltersResponse> {
+  return ipcRenderer.invoke(
+    '@ghostery/adblocker/inject-cosmetic-filters',
+    window.location.href,
+    data,
+  );
 }
 
 if (window === window.top && window.location.href.startsWith('devtools://') === false) {
-  (() => {
-    const enableMutationObserver = ipcRenderer.sendSync('is-mutation-observer-enabled');
+  (async () => {
+    const enableMutationObserver = await ipcRenderer.invoke(
```

Comment:
turning this to async may introduce a noticable delay. 

What is the reason to avoid `ipcRenderer.sendSync`?

Reply:
This will block the renderer until the call is resolved. Definitely not something we would want in our product. Electron itself does suggest to move away from `sendSync` to `invoke` [0].

[0] https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrenderersendsyncchannel-args

Reply:
> turning this to async may introduce a noticable delay.

My understanding of async JavaScript doesn't suggest this. Please correct me if I'm wrong.

Reply:
If this works well in your use-case then we should be good with merging those changes. It is clear that that avoiding blocking of the renderer thread is something beneficial, but we have to be mindful that specifically scriptlet injection is very time sensitive. Some scriptlets have to be injected before the page loads own scripts. Otherwise, some ads wont be blocked.

This is way I strongly suggest to decouple scriplet injection from other cosmetic filter injection as described here https://github.com/ghostery/adblocker/pull/4278#pullrequestreview-2326062789